packages/fuels-types/src/bech32.rs

```diff
@@ -0,0 +1,102 @@
+use crate::errors::Error;
+use bech32::Variant::Bech32m;
+use bech32::{FromBase32, ToBase32};
+use core::str;
+use fuel_tx::Address;
+use std::fmt::{Display, Formatter};
+
+pub const FUEL_BECH32_HRP: &str = "fuel";
+
+#[derive(Debug, Clone)]
+pub struct Bech32Address {
+    plain_addr: Address,
+    encoding: String,
+}
+
+impl Bech32Address {
+    pub fn new(hrp: &str, hash: [u8; 32]) -> Self {
```

Comment:
I think replacing `hash` with `pubkey_hash` or something would make it more clear.

Reply:
What does the `hrp` correspond to? From looking up I know it means "human-readable part" but I'm not sure I understand what it does